File: java-polymorph-json/src/main/java/org/example/ExtendedMessage.java
package org.example;

public class ExtendedMessage extends BaseMessage {
    public String additionalData;

    public ExtendedMessage(String additionalData) {
        super(ExtendedMessage.class);
        this.additionalData = additionalData;
    }

    public ExtendedMessage() {
    }
}

File: csharp-polymorph-json/ExtendedMessage.cs
namespace csharp_polymorph_json
{
    public class ExtendedMessage : BaseMessage
    {
        public string? additionalData { get; set; }

        public ExtendedMessage(string additionalData)
            : base(typeof(ExtendedMessage))
        {
            this.additionalData = additionalData;
        }

        public ExtendedMessage()
        {
        }
    }
}
